experimental/gittuf/rsl.go

```diff
@@ -498,6 +501,70 @@ func (r *Repository) isDuplicateEntry(refName string, targetID gitinterface.Hash
 	return latestUnskippedEntry.TargetID.Equal(targetID), nil
 }
 
+// PropagateChangesFromUpstreamRepositories invokes gittuf's propagation
+// workflow. It inspects the latest policy metadata to find the applicable
+// propagation directives, and executes the workflow on each one.
+func (r *Repository) PropagateChangesFromUpstreamRepositories(ctx context.Context, sign bool) error {
+	if !dev.InDevMode() {
+		slog.Debug("Propagation is only supported in developer mode, skipping check...")
```

Comment:
This is because we may soon be calling this as part of rsl.RecordReferenceEntry etc. We don't want to return an error in those cases.